Add DSTU2 documentation for CarePlan

MaxPhilips opened the pull request:
Do we need to do anything special for the category terminology binding since we are using the argonaut extension?  Should I just include an example for http://argonaut.hl7.org|assess-plan ?

MaxPhilips commented:
@daliboz @jfrhong could you take a look?

joykoduru commented:
@MaxPhilips Can you sneak in a name change on [this line](https://github.com/cerner/fhir.cerner.com/blob/master/lib/resources/example_json/dstu2_examples_relatedperson.rb#L187) to "Peters, Jennifer"?
This is from https://github.com/cerner/fhir.cerner.com/pull/94

MaxPhilips commented:
Update the < and > characters in the medication order examples in this PR too

MaxPhilips commented:
Updated RelatedPerson example:
https://github.com/cerner/fhir.cerner.com/pull/105/commits/0da32762f6e30f94c2fe519aebf70baf83a7b311#diff-b0effe9c78e5301c39fd3f14c92db204

Updated MedicationOrder example:
https://github.com/cerner/fhir.cerner.com/pull/105/commits/0da32762f6e30f94c2fe519aebf70baf83a7b311#diff-d2c45c48de1b501ecb16965e88dc0165

Fisk42 commented on content/dstu2/care-plan.md:
```diff
@@ -0,0 +1,50 @@
+ `patient`    | This or `subject`, or `_id`. Preferred when set simulaneously with `subject`                                     | [`reference`] | Who care plan is for. Example: `patient=Patient/1316024`
+ `subject`    | This or `patient`, or `_id`. Ignored when set simulaneously with `patient`                                       | [`reference`] | Who care plan is for. The `:Patient` modifier is supported. Example: `subject:Patient=1316024`
+ [`_count`]   | N                                                                                                                | [`number`]    | Number of results per page. Defaults to `10`. Capped at `10`.
+
```
Shouldn't status be listed?

MaxPhilips replied:
We did not add a status query parameter

Fisk42 commented:
+1

JFisk42 commented:
+1

cerbs commented on content/dstu2/care-plan.md:
```diff
@@ -0,0 +1,50 @@
+
+    GET /CarePlan?:parameters
+
+### Parameters
```
Do we need to add an Implementation Notes section to call out the modifier element fields we will not return?  Looks like this could apply to CarePlan.activity.detail.status and CarePlan.activity.detail.prohibited

https://github.com/cerner/fhir.cerner.com/blob/master/content/dstu2/device.md#search

MaxPhilips replied:
https://github.com/cerner/fhir.cerner.com/pull/105/commits/5ad8fd719446a3188c9888d6371fff5860e52b40#diff-ccd172a35d4a08ccbd7f98c90c538e79R20

cerbs commented:
👍  after updates

daliboz commented on content/dstu2/care-plan.md:
```diff
@@ -0,0 +1,50 @@
+ Name         | Required?                                                                                                        | Type          | Description
+--------------|------------------------------------------------------------------------------------------------------------------|---------------|-------------------------------------------------------------------------------------
+ `_id`        | This or `patient`/`subject`. If populated: `date`, `patient`, and `subject` are not allowed; `_count` is ignored | [`token`]     | The logical resource id associated with the resource.
+ `date`       | N                                                                                                                | [`date`]      | Time period plan covers. 'ge' and 'le' are supported as prefixes. May be present once to imply a date range, or twice to specify a date range. Partial dates are supported. Example: `date=ge2016-08` `&date=le2017-01-24T12:00:00.000Z`
```
+1

MaxPhilips replied:
Left out that bit

daliboz commented:
Likely need to add something about not supporting the CarePlan.activity field (and thus all the modifier fields under that)

daliboz commented:
The binding on terminology is incorrect. We use multiple different systems, correct?

daliboz commented:
@MaxPhilips I didn't think the RelatedPerson was supposed to be contained - we are fixing that.

MaxPhilips commented on content/dstu2/care-plan.md:
```diff
@@ -0,0 +1,50 @@
+
+ Name         | Required?                                                                                                        | Type          | Description
+--------------|------------------------------------------------------------------------------------------------------------------|---------------|-------------------------------------------------------------------------------------
+ `_id`        | This or `patient`/`subject`. If populated: `date`, `patient`, and `subject` are not allowed; `_count` is ignored | [`token`]     | The logical resource id associated with the resource.
```
Re-wrote the table using Notes
https://github.com/cerner/fhir.cerner.com/pull/105/commits/5ad8fd719446a3188c9888d6371fff5860e52b40#diff-ccd172a35d4a08ccbd7f98c90c538e79L22

MaxPhilips commented:
Added an implementation note for CarePlan.activity: https://github.com/cerner/fhir.cerner.com/pull/105/commits/5ad8fd719446a3188c9888d6371fff5860e52b40#diff-ccd172a35d4a08ccbd7f98c90c538e79R20

I'm unsure on the terminology binding - we populate CarePlan.category only using the Argonaut extension.  So should I remove the binding to http://hl7.org/fhir/ValueSet/care-plan-category and replace with http://argonaut.hl7.org ?

The related person change in this PR is something I overlooked when reviewing #94.  Just updating the patient name to match the id.

daliboz commented:
Link to http://www.fhir.org/guides/argonaut/r2/ValueSet-argo-codesystem.html and specifically call out the assess-plan value.

MaxPhilips commented:
Updated the examples to include better CarePlans

Updated the CarePlan.category description to "Identifies what kind of plan this is. The code will be "assess-plan" from the linked value set." and linked to http://www.fhir.org/guides/argonaut/r2/ValueSet-argo-codesystem.html

Added an implementation note: "The description of the CarePlan is included in the CarePlan.text.div element as XHTML. We will be making changes to increase the usability of the description."

daliboz commented:
+1

MaxPhilips commented:
Squashed commits, merging
